pt/lessons/ecto/changesets.md

```diff
@@ -0,0 +1,290 @@
+---
+version: 1.0.0
+title: Changesets
+---
+
+Para inserir, atualizar ou excluir as informações de um banco de dados, `Ecto.Repo.insert/2`, `update/2` e `delete/2` é necessário um changeset como primeiro parâmetro.
+Mas o que são exatamente changesets?
+
+Uma tarefa comum para quase todos os desenvolvedores é verificar os dados de entrada para possíveis erros — Queremos ter certeza de que os dados estão no estado correto antes de tentarmos usá-los para nossos propósitos.
+
+O Ecto fornece uma solução completa para trabalhar com alteração de dados na forma do módulo `Changeset` e de estruturas de dados. 
+Nesta lição, vamos explorar essa funcionalidade e aprender a verificar a integridade dos dados antes de persisti-los no banco de dados.
+
+{% include toc.html %}
+
+## Criando seu primeiro changeset
+ 
+Vamos ver uma estrutura `%Changeset{}` vazia:
+
+```elixir
+iex> %Ecto.Changeset{}
+#Ecto.Changeset<action: nil, changes: %{}, errors: [], data: nil, valid?: false>
+```
+
+Como você pode ver, tem alguns campos potencialmente úteis, mas estão todos vazios.
+
+Para um changeset ser verdadeiramente útil, quando o criamos, precisamos fornecer um diagrama de como são os dados. 
+Qual o melhor diagrama para nossos dados senão os schemas que criamos que definem nossos campos e tipos?
+
+Vamos ver um schema comum de `User`:
+
+```elixir
+defmodule User do
+  use Ecto.Schema
+
+  schema "users" do
+    field(:name, :string)
+  end
+end
+```
+
+Para criar um changeset usando o schema `User`, vamos usar `Ecto.Changetset.cast/4`:
+
+```elixir
+iex> Ecto.Changeset.cast(%User{name: "Bob"}, %{}, [:name])
+#Ecto.Changeset<action: nil, changes: %{}, errors: [], data: #User<>,
+ valid?: true>
+ ```
+
+O primeiro parâmetro é o dado original - uma stuct `%User{}` vazia neste caso. 
+Ecto é inteligente o suficiente para encontrar o schema baseado na própria estrutura.
+O segundo parâmetro são as alterações que queremos fazer - apenas uma map vazio.
```

Comment:
I believe we have been using masculine genre for maps throughout the lessons: "... apenas **um** map vazio".